cmd/generate.go

```diff
@@ -28,155 +28,165 @@ var generateCmd = &cobra.Command{
 	Short: "Generate a config file from state management",
 	Run: func(cmd *cobra.Command, args []string) {
 		// make sure that we have a token present before trying to make request
-		if config.AccessToken == "" {
+		if conf.AccessToken == "" {
 			// check if OCHAMI_ACCESS_TOKEN env var is set if no access token is provided and use that instead
 			accessToken := os.Getenv("ACCESS_TOKEN")
 			if accessToken != "" {
-				config.AccessToken = accessToken
+				conf.AccessToken = accessToken
 			} else {
 				// TODO: try and fetch token first if it is needed
 				if verbose {
-					fmt.Printf("No token found. Attempting to generate config without one...\n")
+					fmt.Printf("No token found. Attempting to generate conf without one...\n")
 				}
 			}
 		}
 
 		// use cert path from cobra if empty
-		if config.CertPath == "" {
-			config.CertPath = cacertPath
+		if conf.CertPath == "" {
+			conf.CertPath = cacertPath
 		}
 
-		// show config as JSON and generators if verbose
+		// show conf as JSON and generators if verbose
 		if verbose {
-			b, err := json.MarshalIndent(config, "", "  ")
+			b, err := json.MarshalIndent(conf, "", "  ")
 			if err != nil {
-				fmt.Printf("failed to marshal config: %v\n", err)
+				fmt.Printf("failed to marshal conf: %v\n", err)
 			}
 			fmt.Printf("%v\n", string(b))
 		}
 
 		// run all of the target recursively until completion if provided
 		if len(targets) > 0 {
-			RunTargets(&config, args, targets...)
+			RunTargets(&conf, args, targets...)
 		} else {
 			if pluginPath == "" {
 				fmt.Printf("no plugin path specified")
 				return
 			}
 
+			// load the templates to use
+			templates := map[string]generator.Template{}
+			for _, path := range templatePaths {
+				template := generator.Template{}
+				template.LoadFromFile(path)
+				if !template.IsEmpty() {
+					templates[path] = template
+				}
+			}
+
+			params := generator.Params{
+				Templates: templates,
+			}
+
 			// run generator.Generate() with just plugin path and templates provided
-			generator.Generate(&config, generator.Params{
-				PluginPath:    pluginPath,
-				TemplatePaths: templatePaths,
-			})
+			outputBytes, err := generator.Generate(pluginPath, params)
+			if err != nil {
+				log.Error().Err(err).Msg("failed to generate files")
+			}
 
+			// if we have more than one target and output is set, create configs in directory
+			writeOutput(outputBytes, len(targets), len(outputMap))
 		}
 	},
 }
 
 // Generate files by supplying a list of targets as string values. Currently,
-// targets are defined statically in a config file. Targets are ran recursively
+// targets are defined statically in a conf file. Targets are ran recursively
```

Comment:
Another nitpick: We probably can keep "config" here.

Reply:
I agree. I think that was from doing a mass "find and replace" and this was in the crossfire.